### PR TITLE
Make Bytes, BytesIn and ByteOut generic to fix raw type warning https://github.com/OpenHFT/Chronicle-Bytes/issues/358

### PR DESCRIPTION
**Pull request by JerryShea**

### CODE REVIEW

File: chronicle-sandbox/src/main/java/net/openhft/chronicle/engine/client/internal/ClientWiredExcerptAppenderStateless.java [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
+    private final Bytes<?> source = Bytes.elasticByteBuffer();
     private final Wire wire;
     private ChronicleQueue queue;
     private long cid;
     private long lastWrittenIndex = -1;
 
     public ClientWiredExcerptAppenderStateless(ClientWiredChronicleQueueStateless queue,
                                                ClientWiredStatelessTcpConnectionHub hub,
-                                               Function<Bytes, Wire> wireWrapper) {
+                                               Function<Bytes<?>, Wire> wireWrapper) {
```
Comment (minborg) [inline chronicle-sandbox/src/main/java/net/openhft/chronicle/engine/client/internal/ClientWiredExcerptAppenderStateless.java:26]: Not related to this PR but strictly we should declare `Function<? super Bytes<?>, ? extends Wire>`

**Review by minborg — APPROVED** [review]
Remove <?> from comments and javadoc in some cases

File: MigratingToV4.adoc [hunk 2]
```diff
@@ -99,9 +99,9 @@ try (DocumentContext dc = appender.writingDocument()) {
 }
 ----
 
-=== Changes to the Bytes API.
+=== Changes to the Bytes<?> API.
```
Comment (JerryShea) [inline MigratingToV4.adoc]: I think we have to a manual pass through the docs to undo a few of these changes

File: chronicle-sandbox/src/main/java/net/openhft/chronicle/engine/client/internal/ClientWiredExcerptAppenderStateless.java (hunk 1 above)
Comment (peter-lawrey) [reply chronicle-sandbox/src/main/java/net/openhft/chronicle/engine/client/internal/ClientWiredExcerptAppenderStateless.java:26]: Possibly, however I am not sure it's clearer

**Pull request merged**